automatic/blender/update.ps1

```diff
@@ -38,29 +35,32 @@ function global:au_GetLatest {
     $version64 = $version64 -replace $letter,".$num"
   }
 
+  $majorVersion64 = $version64 -split '\.' | Select-Object -First 2
+  $majorVersion64 = $majorVersion64 -join '.'
+
   @{
-    URL64 = Get-ActualUrl $url64
+    URL64 = Get-ActualUrl $version64 $majorVersion64
+    ChecksumType64 = "sha256"
     Version = $version64
+    MajorVersion = $majorVersion64
   }
 }
 
 function Get-ActualUrl() {
-  param([string]$url)
+  param([string]$version, [string]$majorVersion)
 
-  $request = [System.Net.WebRequest]::Create($url)
+  return "https://download.blender.org/release/Blender$majorVersion/blender-$version-windows-x64.msi"
+}
 
-  $response = $request.GetResponse()
+function Get-Checksum() {
+  param([string]$version, [string]$majorVersion, [string]$checksumType)
 
-  $headers = $response.Headers
+  $checksum_file = Invoke-WebRequest -Uri "https://download.blender.org/release/Blender$majorVersion/blender-$version.$checksumType"
```

Comment:
Any chance you can get rid of the hardcoded format?
Either by building the URL based on the MSI download URL, or making another request to the website?

Reply:
Yeah the old URL is a random mirror URL. So there is no guarantee that the URL keeps working now that the user can download this package at some later time.The "hardcoded" url is the official blender file repo containing all Blender versions all the way to the beginning of time. You can find the reference to that URL here: https://www.blender.org/download/previous-versions/.

Reply:
Also this makes checksum file download a lot quicker. Otherwise the au updater would need to download the full binary to generate the checksum.

Reply:
Got it, thanks for that information. In that case I agree with having the hard coded URL, at least for now.